### PR TITLE
Add docker build and push CI workflow

### PR DESCRIPTION
**Pull request by dcroote**
Closes DAO-228 and DAO-161 (the latter once the secrets `DOCKERHUB_USERNAME` and `DOCKERHUB_TOKEN` are saved in the repo Settings).

This workflow builds the Docker image on a push to either `main` or `production`, whereas login & push to Docker Hub occurs only on the `production` branch. The production image is tagged with both `latest` and the SHA in the form of `sha-#######`. 

I've tested this on my fork:
- `main` [Action log](https://github.com/dcroote/api3-dao-dashboard/actions/runs/3457977495/jobs/5771934237): build but no push as expected
- `production` [Action log](https://github.com/dcroote/api3-dao-dashboard/actions/runs/3458031639/jobs/5772041116): login and push to repo - you can see the [tagged images here](https://hub.docker.com/r/dcroote/api3-dao-dashboard/tags). I've tested the `latest` image works correctly for loading the dashboard & connecting a wallet.

### CODE REVIEW

**Review by mcoetzee — APPROVED** [review]
Nice one 🥂

File: .github/workflows/build-docker.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        uses: actions/checkout@v3
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Docker Metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ secrets.DOCKERHUB_USERNAME }}/api3-dao-dashboard
+          tags: |
+            # set latest tag for production branch
```
Comment (Siegrift) [inline .github/workflows/build-docker.yml:23]: The tag should be similar to the other images we have in api3 https://hub.docker.com/u/api3

e.g. `api3/dao-dashboard` (remove the api3 prefix) and maybe full sha hash for the commit tag, like we have [here](https://hub.docker.com/r/api3/airnode-deployer-dev/tags). WDYT?

Comment (dcroote) [reply .github/workflows/build-docker.yml:23]: Great suggestions- I implemented both in the latest commit, can see them here: https://hub.docker.com/r/dcroote/dao-dashboard/tags

**Review by Siegrift — APPROVED** [review]
👍 LGTM.

**Comment by dcroote** [comment]
Whoops sorry for this failing on `main` after merge- I didn't think `main` would need the secrets given the absence of docker login & push, but forgot about the 'secret' being used in the image name. @mcoetzee if you are able to add secrets then adding `DOCKERHUB_USERNAME` as `api3` should fix it.

**Pull request merged**